### ISSUE TITLE
IMPORTANT: master branch in alpha stage - feature freeze for 3.1

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#21490]
Important announcement for all Godot contributors: As of 79c6a83c9, the *master* branch is now in *alpha* stage, which in our release workflow means that it is considered feature-complete. The focus will now switch to bug fixing only, together with last improvements to *existing* features.

We can this stage **feature freeze**, which means that *no new feature PRs will be considered for 3.1*. The current feature set is what 3.1 will be released with, anything else will have to wait for after the 3.1-stable release (in a couple of months) to be reviewed and considered for merging.

As such, we ask all contributors, as far as possible, to *refrain from making feature PRs* (this includes substantial enhancements to existing features, if they involve a lot of code changes/have potential for creating new bugs). Even if you have great ideas of improvement, we strongly suggest to keep them in your local fork for now, and make your PRs once the feature freeze is lifted after the stable release. We won't outright close feature PRs made during the feature freeze, but they will be targeted at the 3.2 milestone anyway, and they might take contributors focus away from bug fixing. Worse, they might end up piled below a huge backlog by the end of the freeze period, and lack visibility (we still haven't fully managed to review all the feature PRs made during 3.0's feature freeze...).

*Existing PRs* slated for the 3.1 milestone can still be reviewed and merged during the alpha stage, up until the beta stage, where *release freeze* will mean that only bug fixes are allowed.

Exceptions to the rule:
- Enhancements to major new features of the 3.1 release (especially usability fixes) may still be considered, if their scope is well defined and they don't risk to introduce regressions. Please discuss with developers on IRC (#godotengine-devel on Freenode) before starting any such work, to ensure that it fits our expectations for what is acceptable during the alpha stage.
- Mono/C# support is still a moving target and can keep being worked on.
- Features that we already discussed on IRC with contributors and approved may still be considered, if they're part of our priorities for 3.1 (again, if unsure, come to IRC for confirmation).

And now, let's focus our efforts on *fixing as many bugs as we can* before 3.1-stable.

Bug reports in 3.1 milestone: https://github.com/godotengine/godot/issues?q=is%3Aopen+is%3Aissue+label%3Abug+milestone%3A3.1

Issues labeled "feature request" or "enhancement" and currently targeting the 3.1 milestone will be reviewed in coming days, and most of them likely moved to the 3.2 milestone, unless considered high priority for 3.1 (e.g. very important enhancements to new 3.1 features).

**Comment by insomniacUNDERSCORElemon** [comment]
Can someone take a look as fixing AA for #10241 / #12840 ? There seems to be a conflict with HDR and AA. This should fix AA at least for polygons on desktop (non-HTML5) platforms.

Although there is more with AA that could be fixed, like things I point out in [this thread](https://www.reddit.com/r/godot/comments/98m7i4/hdr_conflicts_with_aa_also_geometry_experiments/). Things like AA on overlapped shapes (causing fringes on opaque shapes or aliasing on partially transparent ones at specific angles), inner-AA only on partially transparent shapes that is too opaque (and that can cause shimmering), and editor rendering AA-d areas as solid when AA is disabled (similarly to broken AA from HDR+3D framebuffer).

**Comment by aaronfranke** [comment]
You can re-tag [all of my open PRs](https://github.com/godotengine/godot/pulls?q=author%3Aaaronfranke+is%3Aopen) as 3.2 milestone. All are non-vital improvements.

**Comment by aaronfranke** [comment]
Is it about time for 3.1 Alpha 2? There's been a decent amount of fixes since Alpha 1 (1.5 months worth) and the master branch seems fairly stable now.

**Comment by akien-mga** [comment]
Superseded by #24822.
